classes/war/WarManager.php

```diff
@@ -308,4 +386,37 @@ public function handlePatrolDefeat(int $patrol_id) {
         $respawn_time = time() + self::PATROL_RESPAWN_TIME;
         $this->system->db->query("UPDATE `patrols` SET `start_time` = {$respawn_time}, `name` = '{$name}', `ai_id` = {$ai_id}, `tier` = {$tier} WHERE `id` = {$patrol_id}");
     }
+
+    public function handlePatrolWin(int $patrol_id): string
+    {
+        $message = '';
+        // get patrol
+        $result = $this->system->db->query("SELECT * FROM `patrols` WHERE `id` = {$patrol_id} LIMIT 1");
+        $patrol = $this->system->db->fetch($result);
+        $village = VillageManager::getVillageByID($this->system, $patrol['village_id']);
+        // get loot
+        $loot_result = $this->system->db->query("SELECT * FROM `loot` WHERE `user_id` = {$this->user->user_id} AND `claimed_village_id` IS NULL");
+        $loot_result = $this->system->db->fetch_all($loot_result);
+        if ($this->system->db->last_num_rows > 0) {
+            $message .= "<br>Your loot was taken by the enemy patrol and returned to their village.";
+        }
+        $loot_gained = [];
+        foreach ($loot_result as $loot) {
+            // count loot totals
+            if (empty($loot_gained[$loot['resource_id']])) {
+                $loot_gained[$loot['resource_id']] = 1;
+            } else {
+                $loot_gained[$loot['resource_id']] += 1;
+            }
+        }
+        // update village resources
+        foreach ($loot_gained as $resource_id => $count) {
+            $village->addResource($resource_id, $count);
```

Comment:
Hmm - I expect eventually we might want to have controls on how much we add from each type, maybe it would be good to add a type flag to this method like `$village->addResource($resource_id, $count, WarManager::NPC_PATROL_WON)` and then we can control for each way you gather resources, how much it actually adds to the village

But okay to skip, just something to think about for now